internal/configs/parser_config.go

```diff
@@ -370,3 +371,27 @@ var configFileExperimentsSniffBlockSchema = &hcl.BodySchema{
 		{Name: "language"},
 	},
 }
+
+// Sometimes, we manually parse an expression instead of using the hcl library
+// for parsing. In this case we need to handle json configs specially, as the
+// values will be json strings rather than hcl.
+func convertJSONExpressionToHCL(expr hcl.Expression) (hcl.Expression, hcl.Diagnostics) {
```

Comment:
Perhaps this should be moved to the hcl2shim package?  That's where we have historically stored hcl helper functions (though that's a longer discussion).

Reply:
[Done](https://github.com/opentofu/opentofu/pull/1665/commits/95916b88dcbe5976845032350ef6c1c9c25a7f7d)